sway-lsp/src/core/sync.rs

```diff
@@ -23,88 +22,101 @@ pub enum Directory {
+                    } else {
+                        tracing::debug!("Successfully removed temp base dir: {:?}", parent_dir);
+                    }
+                }
+            }
         }
     }
 
-    pub(crate) fn create_temp_dir_from_workspace(
+    pub fn create_temp_dir_from_workspace(
         &self,
-        manifest_dir: &Path,
+        actual_workspace_root: &Path,
     ) -> Result<(), LanguageServerError> {
-        let manifest = PackageManifestFile::from_dir(manifest_dir).map_err(|_| {
-            DocumentError::ManifestFileNotFound {
-                dir: manifest_dir.to_string_lossy().to_string(),
-            }
-        })?;
-
-        // strip Forc.toml from the path to get the manifest directory
-        let manifest_dir = manifest
-            .path()
-            .parent()
-            .ok_or(DirectoryError::ManifestDirNotFound)?;
-
-        // extract the project name from the path
-        let project_name = manifest_dir
+        let root_dir_name = actual_workspace_root
             .file_name()
             .and_then(|name| name.to_str())
-            .ok_or(DirectoryError::CantExtractProjectName {
-                dir: manifest_dir.to_string_lossy().to_string(),
+            .ok_or_else(|| DirectoryError::CantExtractProjectName {
+                dir: actual_workspace_root.to_string_lossy().to_string(),
             })?;
 
-        // Create a new temporary directory that we can clone the current workspace into.
-        let temp_dir = Builder::new()
+        let temp_dir_guard = Builder::new()
             .prefix(SyncWorkspace::LSP_TEMP_PREFIX)
             .tempdir()
             .map_err(|_| DirectoryError::TempDirFailed)?;
 
-        let temp_path = temp_dir
-            .into_path()
-            .canonicalize()
-            .map_err(|_| DirectoryError::CanonicalizeFailed)?
-            .join(project_name);
+        // Construct the path for our specific workspace clone *inside* the directory managed by temp_dir_guard.
+        let temp_workspace_base = temp_dir_guard.path().join(root_dir_name);
+
+        fs::create_dir_all(&temp_workspace_base).map_err(|io_err| {
+            tracing::error!(
+                "Failed to create subdirectory {:?} in temp: {}",
+                temp_workspace_base,
+                io_err
+            );
+            DirectoryError::TempDirFailed
+        })?;
+
+        let canonical_temp_path = temp_workspace_base.canonicalize().map_err(|io_err| {
+            tracing::warn!(
+                "Failed to canonicalize temp path {:?}: {}",
+                temp_workspace_base,
+                io_err
+            );
+            DirectoryError::CanonicalizeFailed
+        })?;
 
         self.directories
-            .insert(Directory::Manifest, manifest_dir.to_path_buf());
-        self.directories.insert(Directory::Temp, temp_path);
+            .insert(Directory::Manifest, actual_workspace_root.to_path_buf());
```

Comment:
nit: canonical paths ensures prefix matching works reliably across OS-specific path representations and symlinks
```suggestion
            .insert(
                Directory::Manifest,
                actual_workspace_root.canonicalize().map_err(|_| DirectoryError::CanonicalizeFailed)?
            );
```

Reply:
[964f7db](https://github.com/FuelLabs/sway/pull/7166/commits/964f7db2aaf796992abe3a9e9bce9a5f37cbda36)